README.md

```diff
@@ -1,120 +1,87 @@
+    // Our chain's account ID type (we can't get away without mentioning it explicitly):
+    AccountId,
+    // We don't track any teleports of `Assets`.
+    Nothing,
+    // We don't track any teleports of `Assets`.
+    CheckingAccount,
+>;
 ```
+Trait definition for PSP34MutateAdapter can be found here-> https://github.com/gitofdeepanshu/polkadot/pull/1
 
-Now, you can obtain the node's session key by sending the following RPC payload.
+## Trait Bounds for Pallet Contracts
+Structure of our new Adapter
 
-```bash
-# send `rotate_keys` request
-$ curl -H 'Content-Type: application/json' --data '{ "jsonrpc":"2.0", "method":"author_rotateKeys", "id":1 }' localhost:9933
-
-# should return a long string of hex, which is your session key
-{"jsonrpc":"2.0","result":"<session key in hex>","id":1}
+```
+impl<
+		Assets: nonfungibles::MutatePSP34<AccountId>,
+		Matcher: MatchesNonFungibles<Assets::CollectionId, Assets::ItemId>,
+		AccountIdConverter: Convert<MultiLocation, AccountId>,
+		AccountId: Clone + Eq + From<[u8;32]>, // can't get away without it since Currency is generic over it.
+		CheckAsset: AssetChecking<Assets::CollectionId>,
+		CheckingAccount: Get<Option<AccountId>>,
+	>
+	PSP34MutateAdapter<
+		Assets,
+		Matcher,
+		AccountIdConverter,
+		AccountId,
+		CheckAsset,
+		CheckingAccount,
+	>
 ```
 
-After this step, you should have a validator node online with a session key for your node.
-For key management and validator rewards, consult our [validator guide online](https://docs.astar.network/build/validator-guide/configure-node).
+In order to make pallet contracts compatible with our new adapter, we have to implement trait `nonfungibles::MutatePSP34<AccountId>` for pallet_contracts.
+Here is the PR for new trait definitions for pallet_contracts: https://github.com/gitofdeepanshu/substrate/pull/1
 
-## Run RPC Tests
+and here is the actual implentation of traits : https://github.com/paritytech/substrate/commit/ab6646e57c8a340d9365833e198f766fefc80ede
 
-RPC tests suite can be run for any release. To run tests go to https://github.com/AstarNetwork/Astar/actions/workflows/rpcTest.yml. Click Run workflow, in the dropdown input the release version tag you want to run the test suite. Then click the green Run workflow button to start the test suite. 
+### MultiAsset Scheme for PSP34 
+To reference a NFT, we need :
 
-![Screenshot from 2022-07-07 15-28-46](https://user-images.githubusercontent.com/874046/177785570-330c6613-237d-4190-bfed-69876209daf6.png)
+a. Smart Contract Address
 
-## Workspace Dependency Handling
+b. Item Id for a specific NFTin collection
 
-All dependencies should be listed inside the workspace's root `Cargo.toml` file.
-This allows us to easily change version of a crate used by the entire repo by modifying the version in a single place.
+```
+MultiAsset {
+id: Concrete( MultiLocation { parents: 0, interior: Junctions::X1(AccountId32 {network: None, id: [u8;32] } ) } ),
+fun: Fungibility::NonFungible::Index(u128)
+```
```

Comment:
This is correct, but it may not be verbose enough (I commented below).

A PSP34 NFT in `pallet-contracts` on `Astar` would need a couple of identifiers.
1. You'd need to specify the parachain Id
2. You'd need to specify which pallet  are you referring to (via pallet index)
3. then come the smart contract Id (address) and collection Id, etc.)

What do you think about that?

Reply:
You are right, we can add them to be more verbose.
The Multilocation I proposed was in context of receiving parachain so I thought we didn't needed much information, but you are right, we should make it more verbose.